frontend/src/pages/components/ConversationPanel.jsx

```diff
@@ -4,43 +4,98 @@ import styles from '../../styles.module.css';
-        fetch('http://localhost:5000/user', {
-            headers: headers,
-            method: 'GET'
-        }).then((response) => {
-            if (response.ok) 
-                return response.json();
-            else 
-                throw response
-            }
-        ).then((response) => {
-            if (response.users) {
-                setUsers(response.users);
-                return response.users;
+        //let chatroomList = []
+        let headers = new Headers();
+        //headers.set('Content-Type', 'application/json');
+        //headers.set('Accept', 'application/json');
+        headers.set('x-access-token', sessionStorage.token);
+        
+        fetch('http://localhost:5000/chat', {
+        headers: headers,
+        method: 'GET',
+        }).then((response) => response.json())
+        .then((response) => {
+            if (response) {
+                let chatroomList = response["chatrooms"];
+
+                let displayChatroom = []
+                //console.log("chatroomList:");
+                //console.log(chatroomList);
+                for (let i = 0; i < chatroomList.length; i++) {
+                    let chatroom = chatroomList[i];
+                    //console.log(chatroom)
+                    if (chatroom["users"].includes(sessionStorage.Username)) {
+                        displayChatroom.push({"label": chatroom["name"], "id": chatroom["public_id"]});
+                    }
+                }
+                //console.log("chats:");
+                //console.log(chats);
+                if (JSON.stringify(chats) != JSON.stringify(displayChatroom)) {
+                    setChats(displayChatroom);
+                    //console.log("chats:");
+                    //console.log(chats);
+                }
+
+
+                return response["chatrooms"];
             } else {
-                console.error(`NO USERS OR SOMETHING IDK`);
+                console.log("couldn't fetch chatrooms");
             }
         });
-    }, [])
+        //let chatroomList = response['chatrooms'];
+
+        //let chatroomList = accurateMockData["chatrooms"];
+    });
 
     const handleAutocomplete = (event, value) => {
         // setUsers(value);
         // fetch endpoint for <id> chat history
-        setMessages(mockData[value]);
+        //setMessages(mockData[value]);
+        setCurrentChat(value);
+        sessionStorage.currentChat = value['id'];
+        console.log("handling autocomplete");
+        
+        //console.log(event);
         console.log(value);
+        console.log("value");
+
+        // call fetch chatid history api here instead of the above setmessages, using value["id"]
+
+        let headers = new Headers();
+        headers.set('Content-Type', 'application/json');
+        headers.set('Accept', 'application/json');
+        headers.set('x-access-token', sessionStorage.token);
+        
+        fetch('http://localhost:5000/chat/' + value["id"], {
```

Comment:
I think the messages from this channel should already be saved in the state as a response from the `join` socket event. Is this doing something different to that?